template/template.go

```diff
@@ -31,28 +31,57 @@ type Template struct {
 	locale   language.Tag
 }
 
-type variable struct {
-	name        *string
-	formatValue *string // TODO(mvilks): "selectValue" to be implemented
-	expression  ast.Expression
+type Result struct {
```

Comment:
`Result` is very abstract and is missing intent. I suggest something like `Resolved`.

Reply:
... or `ResolvedValue`